### PR TITLE
fix(flow): mfp operator missing rows

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?
fix a bug that mfp operator will missing rows if run multiple tims in same tick

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- fix a bug that mfp operator will missing rows if run multiple tims in same tick
- add related tests

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by killme2008** [comment]
Can we test it?

**Comment by discord9** [comment]
> Can we test it?

Unit test are added, but since this bug is not quite yet suitable to be tested using sqlness/tests-intergation crate because it require above certain qps to be triggered, is considering modify tsbs or add a extra script/repo to test it

File: src/flow/src/compute/render/map.rs [hunk 1]
```diff
@@ -115,7 +115,13 @@ fn mfp_subgraph(
 ) {
     let run_mfp = || {
         let all_updates = eval_mfp_core(input, mfp_plan, now, err_collector);
-        arrange.write().apply_updates(now, all_updates)?;
+        // if run mfp in the same tick for multiple times, we need to set last_compaction_time to now-1
+        // so that current rows doesn't get compacted(into nothing) and get missed, that's because mfp's arrangement is future only
+        let mut arrange = arrange.write();
+        if arrange.last_compaction_time() >= Some(now - 1) {
```
Comment (zhongzc) [inline src/flow/src/compute/render/map.rs:121]: So hacky... Why it's unable for mfp to run multiple times in the same tick? Can we choose a better time to do compaction?

**Comment by discord9** [comment]
flow doesn't work real well with sqlness since it's kind of async-ish, and can't update to dest table fast enough before sqlness deem test are done, currently working on that, meanwhile I'm using a python script to locally test flow's functionality

**Review by zhongzc — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]

**Pull request merged**